openbb_platform/providers/biztoc/openbb_biztoc/models/global_news.py

```diff
@@ -0,0 +1,164 @@
+        description="Tag, topic, to filter articles by. Only valid when filter is set to tag.",
+        default=None,
+    )
+    term: Optional[str] = Field(
+        description="Search term to filter articles by. This overrides all other filters.",
+        default=None,
+    )
+
+
+class BiztocGlobalNewsData(GlobalNewsData):
+    """Biztoc Global News Data."""
+
+    __alias_dict__ = {"date": "created", "text": "body", "site": "domain"}
+
+    images: Optional[Dict[str, str]] = Field(
+        description="Images for the article.", alias="img", default=None
+    )
+    favicon: Optional[str] = Field(
+        description="Icon image for the source of the article.", default=None
+    )
+    tags: Optional[List[str]] = Field(description="Tags for the article.", default=None)
+    id: Optional[str] = Field(description="Unique Article ID.", default=None)
+    score: Optional[float] = Field(
+        description="Search relevance score for the article.", default=None
+    )
+
+    @field_validator("date", "updated", mode="before", check_fields=False)
+    def date_validate(cls, v):  # pylint: disable=E0213
+        """Return formatted datetime."""
+        return to_datetime(v).strftime("%Y-%m-%d %H:%M:%S")
+
+
+class BiztocGlobalNewsFetcher(
+    Fetcher[
+        BiztocGlobalNewsQueryParams,
+        List[BiztocGlobalNewsData],
+    ]
+):
+    """Transform the query, extract and transform the data from the Biztoc endpoints."""
+
+    @staticmethod
+    def transform_query(params: Dict[str, Any]) -> BiztocGlobalNewsQueryParams:
+        """Transform the query."""
+        return BiztocGlobalNewsQueryParams(**params)
+
+    @staticmethod
+    def get_news(
+        api_key: str,
+        filter: Literal[
+            "crypto", "hot", "latest", "main", "media", "source", "tag"
+        ] = "latest",
+        source: str = "bloomberg",
+        tag: str = "",
+        term: str = "",
+    ) -> List[Dict]:
+        """Calls the BizToc API and returns the data."""
+
+        term = term.replace(" ", "%20") if term else ""
+        _tags = get_all_tags(api_key)
+        pages = get_pages(api_key)
+        tags = []
+        tag = tag.lower() if tag else ""
+        for page in pages:
+            tags.extend(_tags[page][:])
+
+        _sources = get_sources(api_key)
+        sources = sorted([i["id"] for i in _sources])
+
+        headers = {
+            "X-RapidAPI-Key": f"{api_key}",
+            "X-RapidAPI-Host": "biztoc.p.rapidapi.com",
+            "Accept": "application/json",
+            "Accept-Encoding": "gzip",
+        }
+
+        filter_dict = {
+            "hot": "news/hot",
+            "latest": "news/latest",
+            "crypto": "news/latest/crypto",
+            "main": "news/latest/main",
+            "media": "news/latest/media",
+            "source": f"news/source/{source.lower()}",
+            "tag": f"tag/{tag}",
+        }
+        if filter == "source" and source.lower() not in sources:
```

Comment:
If this is supposed to move to helpers the entire function goes.  Sources and tags are not known in advance because they are dynamic - maybe not each day they change, but at an unknown frequency.

Reply:
part of `get_news()` that was moved.